File: 221102_배열/13300_방 배정/13300.py
import sys

input = sys.stdin.readline

n, k = map(int, input().split())
arr = [[0] * 6 for _ in range(2)]
for i in range(n):
    sex, grade = map(int, input().split())
    arr[sex][grade - 1] += 1

cnt = 0

for i in range(2):
    for j in range(6):
        res = arr[i][j]
        if res == 0:
            continue
        elif res <= k:
            cnt += 1
        elif res > k:
            cnt += res // k
            if res % k != 0:
                cnt += 1

print(cnt)

File: 221102_배열/13300_방 배정/13300.java
import java.io.*;
import java.util.StringTokenizer;


public class Main {
    static int[] arr = new int[12];
    static int ans = 0;

    public static void solve(int k) {
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] > 0) {
                if (arr[i] % k == 0) ans += arr[i] / k;
                else ans += arr[i] / k + 1;
            }
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());


        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            int sex = Integer.parseInt(st.nextToken());
            int grade = Integer.parseInt(st.nextToken());

            // 0 여, 1 남
            if (sex == 0) arr[grade - 1]++;
            else arr[grade + 5]++;
        }

        solve(k);
        bw.write(ans + "\n");
        bw.close();
    }
}